backend/src/dataclasses/track.py

```diff
@@ -1,17 +1,30 @@
-from dataclasses import dataclass
-from typing import List
+from pydantic import BaseModel
+from typing import List, Optional
 
 from src.dataclasses.album import Album
 from src.dataclasses.artist import Artist
+from src.dataclasses.external_ids import ExternalIds
+from src.dataclasses.external_urls import ExternalUrls
+from src.dataclasses.restrictions import Restrictions
 
 
-@dataclass
-class Track:
+class Track(BaseModel):
     album: Album
     artists: List[Artist]
+    available_markets: List[str]
+    disc_number: int
     duration_ms: int
+    explicit: bool
+    external_ids: Optional[ExternalIds] = None
+    external_urls: ExternalUrls
+    href: str
     id: str
+    is_playable: Optional[bool] = None
+    restrictions: Optional[Restrictions] = None
     name: str
     popularity: int
+    preview_url: str
```

Comment:
Similarly, I had to "Optional" this to stop errors appearing:
```suggestion
    preview_url: Optional[str]
```